contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol

```diff
@@ -120,4 +120,32 @@ contract ERC20PositionWithdrawer is ReentrancyGuard
 
         return (tokensWithdrawn, tokensReturned);
     }
+
+    /**
+     * Withdraw WETH tokens from a position, unwrap the tokens, and send back to the trader
+     *
+     * @param  erc20Position  The address of the ERC20Position contract to withdraw from
+     * @return                The amount of ETH withdrawn
+     */
+    function withdrawAsEth(
+        address erc20Position
+    )
+        external
+        returns (uint256)
```

Comment:
So `withdraw` is `nonReentrant`, should this be as well?

It seems like in general we're being a bit inconsistent with what we're making `nonReentrant`. In some places we seem to add it just to be safe even when we don't believe it's necessary, and others not

Reply:
Removing both nonReentrants in general. Only the ERC20Position, the Token, and the ExchangeWrapper/Exchange take over the code execution. Any or all of these are should be well known to the withdrawer and have control of the tokens at some point, so they would be able to "steal" the tokens with or without the ReentrancyGuard

Reply:
Yeah I agree, I just think we need to be consistent and decide on if we're going to keep adding `nonReentrant` to places where we don't believe it's necessary just to be safe. It's fine to remove these though